docs/site/Accessing-http-request-response.md

```diff
@@ -0,0 +1,249 @@
+---
+lang: en
+title: 'Accessing HTTP request and response objects'
+keywords: LoopBack 4.0, LoopBack 4, Node.js, TypeScript, OpenAPI
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Accessing-http-request-response.html
+---
+
+## Accessing HTTP request and response objects
+
+To build REST APIs with LoopBack, it's highly recommended for applications to
+not directly manipulate HTTP (Express) request/response objects. Instead, we
+should use OpenAPI decorators to:
+
+1. Describe operation parameters and body so that such values will be injected
+   as part of request processing.
+
+2. Return a value or promise for the response so that the framework can
+   serialize it into HTTP responses.
+
+For example:
+
+```ts
+import {get} from '@loopback/rest';
+import {inject} from '@loopback/core';
+
+export class PingController {
+  constructor() {}
+
+  // Map to `GET /ping`
+  @get('/ping', {
+    responses: {
+      '200': {
+        description: 'Ping Response',
+        content: {
+          'application/json': {
+            schema: {
+              type: 'object',
+              title: 'PingResponse',
+              properties: {
+                greeting: {type: 'string'},
+                date: {type: 'string'},
+              },
+            },
+          },
+        },
+      },
+    },
+  })
+  ping(@param.query.string('message') msg: string): object {
+    return {
+      greeting: `[Pong] ${msg}`,
+      date: new Date(),
+    };
+  }
+}
+```
+
+There are obvious benefits to stick with the best practice:
+
+1. Allow validation of input/output based on OpenAPI specs.
+2. Allow strongly-typed access to input/output data.
+
+But for some use cases, it's still desirable to access HTTP request and response
+objects, as most Express middleware and routes do. Here are some examples:
+
+- Access request headers that are not described by the OpenAPI spec
+- Access the request stream
+- Set status code of responses
+- Set extra response headers that are not described by the OpenAPI spec
```

Comment:
Can you please mention that we would eventually like to implement first-class support for controllers setting response status code & headers without having to access the Express response object? Here is the GH issue tracking this feature: https://github.com/strongloop/loopback-next/issues/436